core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -104,13 +90,72 @@ object RequestChannel extends Logging {
       }
     }
 
-    trace(s"Processor $processor received request: ${requestDesc(true)}")
+    trace(s"Processor $processor received request: ${summary.getDescription(true)}")
+
+    val summary: RequestSummary = new RequestSummary(
+      processor,
+      header,
+      metrics,
+      context,
+      session,
+      startTimeNanos,
+      requestType = bodyAndSize.request.getClass,
+      description = s"${context.header} -- ${bodyAndSize.request.toString(false)}",
+      detailedDescription = s"${context.header} -- ${bodyAndSize.request.toString(true)}",
+      sizeInBytes = bodyAndSize.size,
+      isFromFollower = bodyAndSize.request match {
+        case r:FetchRequest => r.isFromFollower
+        case _ => false
+      }
+    )
+
+    def releaseBuffer(): Unit = {
+      if (buffer != null) {
+        memoryPool.release(buffer)
+        buffer = null
+      }
+    }
+
+    override def toString = s"Request(processor=$processor, " +
+      s"connectionId=${context.connectionId}, " +
+      s"session=$session, " +
+      s"listenerName=${context.listenerName}, " +
+      s"securityProtocol=${context.securityProtocol}, " +
+      s"buffer=$buffer)"
+  }
+
+  /** Summary of Request without the parsed request **/
+  class RequestSummary(
+    val processor: Int,
+    val header: RequestHeader,
+    val metrics: Metrics,
+    val context: RequestContext,
+    session: Session,
+    startTimeNanos: Long,
+    val requestType: Class[_ <: AbstractRequest],
```

Comment:
We don't need this. As you said elsewhere `ApiKeys` provides the relevant information.